src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -259,38 +251,42 @@ delete_subject_version(server::request_t rq, server::reply_t rp) {
         .value_or(permanent_delete::no)};
     rq.req.reset();
 
+    // Must see latest data to know whether what we're deleting is the last
+    // version
+    co_await rq.service().writer().read_sync();
+
     auto version = invalid_schema_version;
+    bool final_version = false;
     if (ver == "latest") {
+        if (permanent == permanent_delete::yes) {
+            // Permanent deletes on 'latest' not allowed.  They are
+            // transformed into soft deletes
+            permanent = permanent_delete::no;
+        }
```

Comment:
In practice it looks like "latest" should find the latest not soft-deleted one and then continue from there, i.e.e, returning 40407 if it's not soft-deleted.

I think that means this block should just not exist.

Reply:
I was going on this language: (from https://docs.confluent.io/platform/current/schema-registry/develop/api.html)

> For hard-delete of a schema version, you must provide a version number as input; not version:latest, which results in a soft delete only

Is this one of those places the docs disagreed with the implementation?

Reply:
> I was going on this language: (from https://docs.confluent.io/platform/current/schema-registry/develop/api.html)
> 
> > For hard-delete of a schema version, you must provide a version number as input; not version:latest, which results in a soft delete only
> 
> Is this one of those places the docs disagreed with the implementation?

Yeah, looks like it.

Reply:
Updated to match the IRL behaviour.